proposed/dotnet-bootstrapper-cli-experience.md

```diff
@@ -0,0 +1,256 @@
+
+##### `Application Directory`:
+| OS | Directory |
+|----|-----------|
+| **Windows** | `/dnvm` |
+| **Mac** | `/Library/Application\ Support/dnvm/` |
+| **Linux** | `/.local/dnvm` |
+
+### Additional Features
+| Command | Description |
+|---------|-------------|
+| `dotnet install 9` | Installs latest 9.X SDK |
+| `dotnet install 9.0.1xx` | Installs latest 9.0.100 SDK. Equivalent to `rollForward: latestFeature` in `global.json` |
+
+### Nice to Have:
+| Command | Description |
+|---------|-------------|
+| `dotnet install lts` | Installs latest LTS SDK. Updates are impacted as such |
+| `dotnet install sts` | Similar to the above but for STS |
+
+### Future Options
+| Option | Description |
+|--------|-------------|
+| `-f --force` | Install even when the SDK already exists |
+| `-i --install-path` | Install at the specified directory |
+| `-v --verbose` | Show logging messages in stdout |
+| `-a --architecture` | Install for another architecture besides the detected system architecture |
+| `-h --help` | Show help |
+| `-u --url` | Use a custom `cdn` `releases.json` for locked down systems |
+
+`dotnet sdk install <channel>` - noun first equivalent.
+The SDK has recently prioritized `noun` first equivalent command options:
+`https://github.com/dotnet/sdk/issues/9650`
+
+### Additional Possibilities:
+```
+dotnet runtime install <channel>
+dotnet runtime install --aspnetcore
+dotnet runtime install --windowsdesktop
+```
+
+### Implementation Detail
+
+For every install made through `dnvm`, the `global.json` `sdk`, `path`, `rollForward` value is recorded in a shared `dnvm-manifest.json` file, stored at the root of `DOTNET_HOME`.
+
+The `timestamp` of the last modified date of `global.json` is also stored, as we can prevent file reads each time by checking this.
+
+The `dnvm-manifest.json` also stores each `install` along with the `path` and `dependents` of that install, as `global.json` files or as a `dnvm` dependent (fully specified static version).
+
+#### Example dnvm-manifest.json:
+
+```json
+{
+  "globalJsonReferences": [
+    {
+      "filePath": "C:\\projects\\myproject\\global.json",
+      "timestamp": "2025-04-15T14:30:22.1234567Z",
+      "sdk": {
+        "version": "9.0.100",
+        "rollForward": "latestFeature",
+        "allowPrerelease": false,
+        "paths": ["C:\\Program Files\\dotnet\\sdk"]
+      },
+    },
+  ],
+  "installations": [
+    {
+      "version": "9.0.100",
+      "mode": "sdk",
+      "architecture": "x64",
+      "path": "C:\\Users\\username\\dnvm\\installs\\9.0.100",
+      "channel": "unspecified",
+      "dependents": [
+        {
+          "type": "globalJson",
+          "path": "C:\\projects\\myproject\\global.json"
+        }
+      ]
+    },
+    {
+      "version": "9.0.200-preview.1",
+      "mode": "sdk",
+      "architecture": "x64",
+      "path": "C:\\Users\\username\\dnvm\\installs\\9.0.200-preview.1",
```

Comment:
Where would installations for non-system architectures go?

For example, `dotnet install -a arm64 9.0.200-preview.1` assumedly would not install into the same directory as the x64 build?

Reply:
Keep in mind that while it's not common, a single OS installation can be multi-architecture-capable. I believe macOS at the moment is capable of a single installation being able to boot into both x86_64 and arm64 depending on what CPU is present - practically, probably when installed to an external disk or as a VM. I suspect Windows can't do the same but I wouldn't be surprised if Linux could.

This edge case may not be a practical concern, or it could be explicitly unsupported, but it is quite possible then that I could install x64 on an arm64 machine, then boot into x64 on the same disk, and it would look in an unprefixed directory and load arm64 .NET on an x64 host 😄.